src/rulesets/LinearAlgebra/blas.jl

```diff
@@ -37,29 +37,34 @@ end
 
 function frule((_, Δx), ::typeof(BLAS.nrm2), x)
     Ω = BLAS.nrm2(x)
-    return Ω, sum(Δx .* @thunk(x * inv(Ω)))
+    s = ifelse(iszero(Ω), one(Ω), Ω)
+    ∂Ω = if eltype(x) <: Real
+        BLAS.dot(x, Δx) / s
+    else
+        sum(zip(x, Δx)) do (xi, Δxi)
+            return _realconjtimes(xi, Δxi)
+        end / s
```

Comment:
Something like this is probably a bit cleaner
```suggestion
        sum(y -> _realconjtimes(y...), zip(x, Δx)) / s
```